### DISCUSSION TITLE
AMS reader connected to a Kaifa MA309 from Netz-noe

### DISCUSSION THREAD

**Discussion by Noschvie** [UtilitechAS/amsreader-firmware#407] [Show and tell]
Find below a screenshot running AMS reader connected to a Kaifa MA309 from [Netz-noe](https://smartmeter.netz-noe.at), Austria.
It's the same interface as for the Sagemcom T210-D.

![grafik](https://user-images.githubusercontent.com/48217146/213872703-14ef57ad-3edd-4206-b6da-39b8275e8c44.png)

**Comment by ArnieO** [comment]
Thank you for the notification! 
What hardware are you running this on?

**Reply by Noschvie** [reply]
Using an ESP32 nodemcu with the [M-Bus Slave Click](https://www.mikroe.com/m-bus-slave-click).
Tested with an ESP8266 successfully too, using an ESP32 C3mini wasn't successful.

**Comment by Noschvie** [comment]
Hello
I'm running version 2.2.23 at [this (480)](https://github.com/UtilitechAS/amsreader-firmware/discussions/480) hardware.

/data.json provides this
`"ic":14866.776,
"ec":12866.487,
`
and list 4 this
`"tPI":14867.49,
"tPO":12866.49,
`

Why are the values of format "list 4" just with 2 decimal places?
Thanks!

**Reply by ArnieO** [reply]
@Noschvie : Which MQTT version is this?
@gskjold : Is the firmware doing a rounding in the MQTT payload?

**Reply by Noschvie** [reply]
Payload is configured as JSON

![grafik](https://github.com/UtilitechAS/amsreader-firmware/assets/48217146/a276611c-1e5b-4d0e-8a16-df6b1f955ae9)

**Reply by ArnieO** [reply]
Thank you - and waiting for @gskjold to check the code.

**Reply by gskjold** [reply]
I guess we could adjust this, it was originally limited to the decimal points needed for display it in the GUI.

**Reply by gskjold** [reply]
Sorry, I misread, this is actually the MQTT payload, will fix!

**Reply by Noschvie** [reply]
Will this fix be included in the upcoming release? Thanks!

**Comment by Noschvie** [comment]
Working on a flow with Node-RED to store the sensor values into InfluxDB.

The device ins running v2.2.23 and MQTT payload is configured as "JSON".
[JSON mode](https://github.com/UtilitechAS/amsreader-firmware/wiki/MQTT-configuration#json-mode) says "Sends all data on single topic with different format." 
For this reason messages with topic "System" have to been filtered and destroyed. Just message with topic "List x" can be used as sensor values.
Suggest to have different topics for these messages.

**Comment by shorshiii** [comment]
Hey @Noschvie , i have a Sagemcom T210-D in NÖ and i am VERY new to all of this, i have almost 0 clue what im doing. I need help. I bought a SlimmeLezer+ to read the signal from the P1 Port, i got my decryption key from EVN, and after some detours (esp-link???) i managed to flash AMS reader 2.4.0 onto it. Now i cant figure out how to make it work, HAN is always red, no matter what meter settings i put in. It just keeps saying "Unknown Data received" What am i doing wrong

//edit: im just now reading on the SlimmeLezer website it doesnt work on P1 ports that are mbus. The T210-D from EVN is mbus. fuck. 

//edit2: i also own a "Everhome EcoTracker IR" (got it from a friend) which attaches to the magnet ring ... Do you know if i can make that one work somehow?

**Reply by Noschvie** [reply]
@shorshiii sorry, I'm not familiar with this hardware. I'm using @dbeinder's module with these settings

![grafik](https://github.com/user-attachments/assets/c8a00d3a-ba21-4075-a4ee-7c48f164fb53)

**Reply by dbeinder** [reply]
@Noschvie Are you using my HW on a Sagecom T210-D without external power? Is it stable, and firmware update works? I had thought only Kaifa provides enough power on the MBUS port

And my module requires "inverted" serial, unless you are still on an old firmware version.

**Reply by Noschvie** [reply]
> Are you using my HW on a Sagecom T210-D without external power?

Yes, I do.

Below are my notes on this.

```
Sagemcom Drehstromzähler T210-D
OTA Upgrade auf 2.3.5 gestartet und ok.
HAN Verbindung kann nicht hergestellt werden.
https://esp.huhn.me/
Mit esp.huhn.me Version 2.2.28 installiert, HAN Connection ok.
OTA Upgrade von 2.2.28 to 2.3.8 gestartet und ok, keine HAN Connection möglich und not ok.
[Discussions #835](https://github.com/UtilitechAS/amsreader-firmware/discussions/835)
```

**Reply by dbeinder** [reply]
But after checking "inverted" it works, right? https://github.com/UtilitechAS/amsreader-firmware/issues/892#issuecomment-2564008058

**Reply by Noschvie** [reply]
Yes it is, sorry for the confusion.

**Comment by Noschvie** [comment]
@martin1neu have a look to this discussion.

**Comment by martin1neu** [comment]
Hi @Noschvie 
thanks for the link. but i'm not exactly sure what you're trying to tell me. 
Do you mean that it might be the hardware or somthing with "invertet" serial?

**Reply by Noschvie** [reply]
I have to check the "inverted" box.

**Reply by martin1neu** [reply]
Thank you for your info. I dosn´t find any option in Tasmota to invert the signal.
After many hours search, trial & error now I have take a second ESP to read out the second smartmeter.

not smart, but it runs
